### PR TITLE
MAINT: Add warnings if pivot value is close to tolerance in linprog(method='simplex')

### PR DESCRIPTION
**Pull request by Kai-Striega**
Implementation of the first steps discussed in #9076.

I've implemented the reporting as warnings instead of a status/message, in theory (or depending on how the tolerance / problem is formed) it may still be on the tableau and maintain feasibility. The definition of "close" to tolerance has been set to be within 4 orders of magnitude (rtol of 1e4) this is really a guess. It catches #5400, #7237, #8174 comfortably and could possibly be fine tuned.

I'm not sure if the implementation of my tests are as well structured as they could be. All fail before and pass after the changes but just feel a bit off. #7237 returns the correct solution when using Bland's rule and has been added as a test.

### CODE REVIEW

**Review by mdhaber — APPROVED** [review]
@Kai-Striega I don't see anything wrong. Is there something specific you want me to double-check?
There are some PEP8 issues to fix, though.

File: scipy/optimize/_linprog.py [hunk 1]
```diff
@@ -218,6 +218,45 @@ def _pivot_row(T, basis, pivcol, phase, tol=1.0E-12, bland=False):
...
+        if irow != pivrow:
+            T[irow] = T[irow] - T[pivrow] * T[irow, pivcol]
+
+    # The selected pivot should never lead to a pivot value less than the tol.
+    if np.isclose(pivval, tol, atol=0, rtol=1e4):
+        message = (
+            "The pivot operation produces a pivot value of:{0: .1e}. "
+            "Being only slightly greater than the set tolerance{1: .1e}. "
+            "This may lead to issues regarding the numerical stability of "
+            "the simplex method. Increasing the tolerance, changing the pivot "
```
[the remark this answers is not in the document]
Comment (Kai-Striega) [reply scipy/optimize/_linprog.py]: Yes, in theory. Having this as the first option probably wasn't the best decision and increasing the tolerance should be avoided anyway. I've change the order of the suggestions to:

- Removing redundant constraints.
- Using Bland's rule.
- Increasing the tolerance

File: scipy/optimize/_linprog.py [hunk 2]
```diff
@@ -218,6 +218,45 @@ def _pivot_row(T, basis, pivcol, phase, tol=1.0E-12, bland=False):
     return True, min_rows[0]
 
 
+def _apply_pivot(T, basis, pivrow, pivcol, tol=1e-12):
```
[the remark this answers is not in the document]
Comment (Kai-Striega) [reply scipy/optimize/_linprog.py]: The change was basically to make sure I didn't forget to change the error message in both places. Other than that the code is copy and pasted from the two sections.

**Comment by Kai-Striega** [comment]
@mdhaber do you think this is ready to be merged? I can then take a look at merging the simplex and IP methods.

**Comment by mdhaber** [comment]
Yeah if there was nothing specific you wanted me to double check. You mentioned something felt uneasy.

**Pull request merged**